### ISSUE TITLE
`@weave.op()` does not work with Pydantic `BaseModel` functions or class variables

### ISSUE THREAD

**Issue by willbakst** [wandb/weave#1364]
## `@weave.op()` failure with `BaseModel` internal function

The code below fails the the error `AttributeError: 'ObjectRecord' object has no attribute 'format'`

```python
import weave
from pydantic import BaseModel


class Book(BaseModel):
    title: str
    author: str

    def format(self) -> str:
        return f"{self.title} by {self.author}"


@weave.op()
def my_func(book: Book) -> str:
    return book.format()


weave.init("scratch-test")
book = Book(title="The Name of the Wind", author="Patrick Rothfuss")
print(my_func(book))
```

## `@weave.op()` failure with `BaseModel` class variable

The code below fails with the error `AttributeError: 'ObjectRecord' object has no attribute 'class_var'`

```python
from typing import ClassVar

import weave
from pydantic import BaseModel


class Book(BaseModel):
    title: str
    author: str

    class_var: ClassVar[str] = "class_var"


@weave.op()
def my_func(book: Book) -> str:
    return book.class_var


weave.init("scratch-test")
book = Book(title="The Name of the Wind", author="Patrick Rothfuss")
print(my_func(book))
```

**Comment by willbakst** [comment]
Would you be open to a PR if I look into fixing this?

Any reference material I should look at before submitting one?

**Comment by willbakst** [comment]
The culprit: https://github.com/wandb/weave/blob/41b5ad32303764585871da792f038510e40ac44e/weave/trace/object_record.py#L39

Looks like this only grabs model fields and doesn't grab any other information, which explains the missing functions and class variables.

**Comment by willbakst** [comment]
Due to how pydantic BaseModel handling is implemented, `@weave.op()` removes everything except for the fields. This means built-in methods like `model_dump` are also no longer present.

I will try to update the code to handle porting over everything.

**Comment by willbakst** [comment]
Fixed with #1390 

Thanks!!
